### PR TITLE
[Products][Taxon] Allowing deleting a taxon that is a main taxon of a product

### PR DESCRIPTION
**Pull request by Rafikooo**
Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         |  1.13                   |
| Bug fix?        | kind of                                                      |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                      |
| Deprecations?   | no |
| Related tickets | https://github.com/Sylius/Sylius/pull/14791  |
| License         | MIT                                                          |

This PR changes (or more accurately unifies) the way we can delete taxon in Sylius.
Previously, we could delete a taxon that is not `main_taxon` for any product, and we could not delete a taxon in the opposite case.

This PR allows us to remove taxon that is `main_taxon` for some product by setting the relationship to `NULL`.

Alternatively, we could prevent the deletion of taxon in both cases. Let's bring this up for discussion 💬

### CODE REVIEW

**Comment by jakubtobiasz** [comment]
It seems the JS build fails are not the random ones :0.

File: features/taxonomy/managing_taxons/deleting_taxon.feature [hunk 1]
```diff
@@ -32,6 +32,14 @@ Feature: Deleting a taxon
...
         Then I should be notified that I cannot delete a menu taxon of any channel
 
+    @ui @javascript @api
+    Scenario: Deleting a taxon that is a main taxon of a product
+        And the store classifies its products as "T-Shirts"
+        And the store has a product "T-Shirts PHP"
+        And this product has a main taxon "T-Shirts"
+        When I delete taxon named "T-Shirts"
+        Then the taxon named "T-Shirts" should no longer exist in the registry
+
```
Comment (jakubtobiasz) [inline features/taxonomy/managing_taxons/deleting_taxon.feature]: Maybe let's add what happens with the product 🤔?

**Review by jakubtobiasz — DISMISSED** [review]

File: features/taxonomy/managing_taxons/deleting_taxon.feature (hunk 1 above)
Comment (Rafikooo) [reply features/taxonomy/managing_taxons/deleting_taxon.feature]: Good catch!

**Review by NoResponseMate — DISMISSED** [review]

**Review by jakubtobiasz — APPROVED** [review]

File: src/Sylius/Bundle/CoreBundle/Migrations/Version20230426154358.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Sylius\Bundle\CoreBundle\Doctrine\Migrations\AbstractMigration;
+
+final class Version20230426154358 extends AbstractMigration
```
Comment (coldic3) [inline src/Sylius/Bundle/CoreBundle/Migrations/Version20230426154358.php:19]: Could you merge it into `Version20230426153930`?

Comment (Rafikooo) [reply src/Sylius/Bundle/CoreBundle/Migrations/Version20230426154358.php:19]: Both migrations extend different AbstractMigration class, and I'm not quite sure how they can be merged

Comment (coldic3) [reply src/Sylius/Bundle/CoreBundle/Migrations/Version20230426154358.php:19]: Oh I see, one is for PostgreSQL :+1:

Comment (NoResponseMate) [reply src/Sylius/Bundle/CoreBundle/Migrations/Version20230426154358.php:19]: image

**Review by NoResponseMate — APPROVED** [review]

**Comment by jakubtobiasz** [comment]
Thank you, @Rafikooo!

**Pull request merged**